go.mod

```diff
@@ -0,0 +1,86 @@
+	github.com/gorilla/handlers v0.0.0-20150720190736-60c7bfde3e33
+	github.com/gorilla/mux v1.4.0 // indirect
+	github.com/gregjones/httpcache v0.0.0-20170728041850-787624de3eb7 // indirect
+	github.com/hashicorp/golang-lru v0.0.0-20160207214719-a0d98a5f2880
+	github.com/imdario/mergo v0.3.5 // indirect
+	github.com/json-iterator/go v0.0.0-20180701071628-ab8a2e0c74be // indirect
+	github.com/marstr/guid v1.1.1-0.20170427235115-8bdf7d1a087c // indirect
+	github.com/matttproud/golang_protobuf_extensions v1.0.1 // indirect
+	github.com/mitchellh/mapstructure v0.0.0-20150528213339-482a9fd5fa83 // indirect
+	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
+	github.com/modern-go/reflect2 v1.0.1 // indirect
+	github.com/ncw/swift v1.0.24-0.20160519162137-ce444d6d47c5 // indirect
+	github.com/opencontainers/go-digest v1.0.0-rc1.0.20180430190053-c9281466c8b2
+	github.com/opencontainers/image-spec v1.0.1-0.20180918080442-7b1e489870ac // indirect
+	github.com/opencontainers/runc v1.0.0-rc5.0.20180920170208-00dc70017d22 // indirect
+	github.com/openshift/api v3.9.1-0.20190424152011-77b8897ec79a+incompatible
+	github.com/openshift/client-go v0.0.0-20190412095722-0255926f5393
+	github.com/openshift/library-go v0.0.0-20190429145318-080e8da3a666
+	github.com/pborman/uuid v0.0.0-20150603214016-ca53cad383ca
+	github.com/peterbourgon/diskv v2.0.1+incompatible // indirect
+	github.com/pkg/errors v0.8.0 // indirect
+	github.com/pkg/profile v1.2.2-0.20180809112205-057bc52a47ec // indirect
+	github.com/prometheus/client_golang v0.8.1-0.20170531130054-e7e903064f5e
+	github.com/prometheus/client_model v0.0.0-20150212101744-fa8ad6fec335 // indirect
+	github.com/prometheus/common v0.0.0-20170615072016-3233b24a3671 // indirect
+	github.com/prometheus/procfs v0.0.0-20170519190837-65c1f6f8f0fc // indirect
+	github.com/satori/go.uuid v1.2.1-0.20180103174451-36e9d2ebbde5 // indirect
+	github.com/sirupsen/logrus v1.0.4-0.20170822132746-89742aefa4b2
+	github.com/spf13/pflag v1.0.1 // indirect
+	golang.org/x/crypto v0.0.0-20180808211826-de0752318171 // indirect
+	golang.org/x/net v0.0.0-20180124060956-0ed95abb35c4 // indirect
+	golang.org/x/oauth2 v0.0.0-20170412232759-a6bd8cefa181 // indirect
+	golang.org/x/sys v0.0.0-20171031081856-95c657629925 // indirect
+	golang.org/x/text v0.0.0-20170810154203-b19bf474d317 // indirect
+	golang.org/x/time v0.0.0-20161028155119-f51c12702a4d // indirect
+	google.golang.org/api v0.0.0-20160322025152-9bf6e6e569ff // indirect
+	google.golang.org/appengine v0.0.0-20160301025000-12d5545dc1cf // indirect
+	google.golang.org/cloud v0.0.0-20151119220103-975617b05ea8 // indirect
+	google.golang.org/genproto v0.0.0-20170731182057-09f6ed296fc6 // indirect
+	google.golang.org/grpc v1.13.0 // indirect
+	gopkg.in/inf.v0 v0.9.0 // indirect
+	gopkg.in/yaml.v2 v2.2.1
+	k8s.io/api v0.0.0-20190222213804-5cb15d344471
+	k8s.io/apimachinery v0.0.0-20190221213512-86fb29eff628
```

Comment:
see the same thing in `glide.yaml` but I didn't realize that image-registry went off base k8s and not
https://github.com/openshift/kubernetes-*

If you have the cycles @dmage, for my own education, what is the history / rationale there?

Reply:
@gabemontero as some point we needed to have a critical fix that had already been backported to all upstream client-go branches, but our client-go didn't have this fix and we weren't confident about rebasing our client-go. The master team suggested to use the upstream branch instead of our fork to get this fix. And today, as I can see, other components (cluster-version-operator, cluster-openshift-apiserver-operator) don't use our fork.

Reply:
gotch - cool, thanks for the background @dmage

Reply:
👍 to @dmage . I think in general the rule is "use upstream" unless our fork has a carry or backport that is critical for security/functionality.